web/controllers/courseController/index.ts

```diff
@@ -2806,7 +2807,12 @@ async function groupMail(student: Student, courseId: number, subcourseId: number
     }
 
     try {
-        for (let participant of subcourse.participants) {
+        const addressees = [];
+        for (let rawAddressee of rawAddressees) {
+            const pupil = await entityManager.findOne(Pupil, { wix_id: rawAddressee});
```

Comment:
What happens if the pupil does not exist? Maybe a `findMany({ wix_id: rawAdressees })` could be used here to send one DB request instead of n ...

Reply:
Did you mean `entityManager.find()` instead of `findMany`? TypeORM doesn't have a findMany function.... Unfortunately, the docs are pretty sparse about this topic....

When I try to use `entityManager.find(Pupil, {wix_id: rawAddressees})`, I get the following TS error:
```
No overload matches this call.
  Overload 1 of 2, '(entityClass: EntityTarget<Pupil>, options?: FindManyOptions<Pupil>): Promise<Pupil[]>', gave the following error.
    Argument of type '{ wix_id: string[]; }' is not assignable to parameter of type 'FindManyOptions<Pupil>'.
      Object literal may only specify known properties, and 'wix_id' does not exist in type 'FindManyOptions<Pupil>'.
  Overload 2 of 2, '(entityClass: EntityTarget<Pupil>, conditions?: FindConditions<Pupil>): Promise<Pupil[]>', gave the following error.
    Type 'string[]' is not assignable to type 'string | FindOperator<string>'.
      Type 'string[]' is not assignable to type 'string'.ts(2769)
Pupil.ts(28, 5): The expected type comes from property 'wix_id' which is declared here on type 'FindConditions<Pupil>'
```

Maybe the QueryBuilder could be used for this?

Reply:
Ah, `.find({ wix_id: In(rawAdresses) })` then ... Yeah Typeorm's docs are really a nightmare.